tests/iface_namingmode/test_iface_namingmode.py

```diff
@@ -960,33 +961,63 @@ def setup_check_topo(self, setup, tbinfo):
         Returns:
-            spine_ports: dictionary containing lists of aliases and names
+            static_route_intf: dictionary containing lists of aliases and names
             of the spine ports
         """
-        minigraph_neighbors = setup['minigraph_facts']['minigraph_neighbors']
-        spine_ports = dict()
-        spine_ports['interface'] = list()
-        spine_ports['alias'] = list()
-
-        for key, value in list(minigraph_neighbors.items()):
-            if (key in setup['physical_interfaces']
-                    and ('T2' in value['name'] or (tbinfo['topo']['type'] == 't2' and 'T3' in value['name']))):
-                spine_ports['interface'].append(key)
-                spine_ports['alias'].append(setup['port_name_map'][key])
+        duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
+        static_route_intf = dict()
+        static_route_intf['interface'] = list()
+        static_route_intf['alias'] = list()
+        gw_ip_list = []
 
-        if not spine_ports['interface']:
+        if not setup['physical_interfaces']:
             pytest.skip('No non-portchannel member interface present')
 
-        logger.info('spine_ports:\n{}'.format(spine_ports))
-        return spine_ports
+        for mg_intf in setup['minigraph_facts'][u'minigraph_interfaces']:
+            if mg_intf[u'attachto'] == setup['physical_interfaces'][0]:
+                dev = mg_intf[u'attachto']
+                namespace = setup['minigraph_facts']['minigraph_neighbors'][dev]['namespace']
+                gw_ip = mg_intf['peer_addr']
+                if ipaddress.ip_address(gw_ip).version == 4:
+                    ip_version = ''
+                    dst_ip = '192.168.1.1'
+                else:
+                    ip_version = '-6'
+                    dst_ip = '::/0'
+                if namespace:
+                    duthost.shell("ip netns exec {} ip {} route add {}  via {} dev {}".
+                                  format(namespace, ip_version, dst_ip, gw_ip, dev))
+                else:
+                    duthost.shell("ip {} route add {}  via {} dev {}".format(ip_version, dst_ip, gw_ip, dev))
+                static_route_intf['interface'].append(dev)
+                static_route_intf['alias'].append(setup['port_name_map'][dev])
+                gw_ip_list.append((gw_ip, namespace))
+
+        yield static_route_intf
+
+        for gw_ip_ns in gw_ip_list:
+            gw_ip, namespace = gw_ip_ns
+            if ipaddress.ip_address(gw_ip).version == 4:
+                ip_version = ''
+                dst_ip = '192.168.1.1'
+            else:
+                ip_version = '-6'
+                dst_ip = '::/0'
+
+            if namespace:
+                duthost.shell("ip netns exec {} ip {} route del {} via {}".
+                              format(namespace, ip_version, dst_ip, gw_ip))
+            else:
+                duthost.shell("ip {} route del {} via {}".
```

Comment:
For ipv6 will this remove the default route learn via BGP as well?

Reply:
no, this will only delete the static route that was added.
```
admin@device:~$ show ipv6 route ::/0
Routing entry for ::/0
  Known via "bgp", distance 20, metric 0, best
  Last update 00:12:36 ago
  * fc00::2, via PortChannel102
  * fc00::a, via PortChannel105
  * fc00::12, via PortChannel108
  * fc00::1a, via PortChannel1011
  * fc00::22, via PortChannel1014
  * fc00::2a, via PortChannel1017
  * fc00::32, via PortChannel1020
  * fc00::3a, via PortChannel1023

admin@device:~$ sudo ip -6 route add ::/0  via fc00::6e dev Ethernet108
admin@device:~$ show ipv6 route ::/0
Routing entry for ::/0
  Known via "kernel", distance 0, metric 1024, best
  Last update 00:00:04 ago
  * fc00::6e, via Ethernet108

Routing entry for ::/0
  Known via "bgp", distance 20, metric 0
  Last update 00:12:43 ago
  * fc00::2, via PortChannel102
  * fc00::a, via PortChannel105
  * fc00::12, via PortChannel108
  * fc00::1a, via PortChannel1011
  * fc00::22, via PortChannel1014
  * fc00::2a, via PortChannel1017
  * fc00::32, via PortChannel1020
  * fc00::3a, via PortChannel1023

admin@device:~$ sudo ip -6 route del ::/0  via fc00::6e
admin@device:~$ show ipv6 route ::/0
Routing entry for ::/0
  Known via "bgp", distance 20, metric 0, best
  Last update 00:13:24 ago
  * fc00::2, via PortChannel102
  * fc00::a, via PortChannel105
  * fc00::12, via PortChannel108
  * fc00::1a, via PortChannel1011
  * fc00::22, via PortChannel1014
  * fc00::2a, via PortChannel1017
  * fc00::32, via PortChannel1020
  * fc00::3a, via PortChannel1023

```

Reply:
@arlakshm modified to use a different ipv6 destination ip to avoid changing default route.